Sources/TuistCore/ValueGraph/ValueGraphDependency.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+import TSCBasic
+
+enum ValueGraphDependency: Hashable {
+    /// A dependency that represents a pre-compiled .xcframework.
+    case xcframework(
+        path: AbsolutePath,
+        infoPlist: XCFrameworkInfoPlist,
+        primaryBinaryPath: AbsolutePath,
+        linking: BinaryLinking
+    )
+
+    /// A dependency that represents a pre-compiled framework.
+    case framework(
+        path: AbsolutePath,
+        dsymPath: AbsolutePath?,
+        bcsymbolmapPaths: [AbsolutePath],
+        linking: BinaryLinking,
+        architectures: [BinaryArchitecture]
+    )
+
+    /// A dependency that represents a pre-compiled library.
+    case library(
+        path: AbsolutePath,
+        publicHeaders: AbsolutePath,
+        architectures: [BinaryArchitecture],
+        linking: BinaryLinking,
+        swiftModuleMap: AbsolutePath?
+    )
+
+    /// A dependency that represents a package product.
+    case packageProduct(product: String, path: AbsolutePath)
```

Comment:
similarly `path` is the first argument for the other dependencies

Reply:
Agree. I'll update this one too.